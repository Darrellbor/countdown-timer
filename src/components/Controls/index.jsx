//@flow
import React, { type Node, useState } from "react";

type ControlsProps = {
  theme: string,
  icon: Node,
  hoverIcon: Node,
  text: string,
  textPos: string,
  controlClicked: Boolean,
  onClick: Function,
};

const Controls = (props: ControlsProps) => {
  const [hover, toggleHover] = useState(false);

  return (
    <div
      className={`Controls ${
        props.theme === "Dark"
          ? "Controls--dark"
          : "Controls--light"
      } ${props.controlClicked && "Controls__clicked"} `}
      onMouseOver={() => toggleHover(true)}
      onMouseOut={() => toggleHover(false)}
      onClick={props.onClick}
    >
      <div className="Controls__inner">
        {props.textPos === "left" && <h5>{props.text}</h5>}
        <span>{hover ? props.hoverIcon : props.icon}</span>
        {props.textPos === "right" && <h5>{props.text}</h5>}
      </div>
    </div>
  );
};

Controls.defaultProps = {
  textPos: "right",
};

export default Controls;
