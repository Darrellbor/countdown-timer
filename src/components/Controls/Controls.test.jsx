import React from "react";
import { shallow } from "enzyme";
import sinon from "sinon";

import Controls from "./";
import Pause from "../Icons/Pause";

describe("Controls Component", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Controls
        theme="Dark"
        icon={<Pause />}
        hoverIcon={<Pause hover={true} />}
        text="Pause Timer"
        textPos="right"
        controlClicked={false}
        onClick={() => {}}
      />
    );
  });

  test("renders correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("calls prop onClick when clicked", () => {
    const spy = sinon.spy();
    wrapper = shallow(<Controls onClick={spy} />);

    wrapper.find(".Controls").simulate("click");

    expect(spy.calledOnce).toBe(true);
  });

  test("check if theme class is applied", () => {
    expect(wrapper.exists(".Controls--dark")).toBeTruthy();
  });

  test("Check if text is displayed", () => {
    expect(wrapper.find("h5").text()).toBe("Pause Timer");
  });

  test("Check if text is displayed on the left", () => {
    wrapper = shallow(<Controls textPos="left" />);
    expect(wrapper.exists("#Controls__inner__left")).toEqual(true);
  });

  test("Check if text is displayed on the right", () => {
    expect(wrapper.exists("#Controls__inner__right")).toEqual(true);
  });

  test("Check if icon is displayed", () => {
    expect(
      wrapper.find("span").contains(<Pause hover={false} theme="Dark" />)
    ).toEqual(true);
  });

  test("Check if icon is displayed on hover", () => {
    wrapper.find(".Controls").simulate("mouseover");
    expect(
      wrapper.find("span").contains(<Pause hover={true} theme="Dark" />)
    ).toEqual(true);
  });
});
