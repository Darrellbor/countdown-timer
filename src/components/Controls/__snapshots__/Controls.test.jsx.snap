// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls Component renders correctly 1`] = `
<div
  className="Controls Controls--dark false "
  onClick={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
>
  <div
    className="Controls__inner"
  >
    <span>
      <Pause
        hover={false}
        theme="Dark"
      />
    </span>
    <h5
      id="Controls__inner__right"
    >
      Pause Timer
    </h5>
  </div>
</div>
`;
